### PR TITLE
Hide fine-grained Zipkin traces behind system property

### PR DESCRIPTION
**Pull request by wiwa**
This change introduces a way to hide some verbose fine-grained zipkin traces with a `bloop.trace.verbose` system property which defaults to `false`. This reduces the number of spans in a given trace around the neighborhood of 80% by culling some traces when the amount of time spent in those traced operations is very low, think microseconds or a few milliseconds in a compile taking a few seconds.

@olafurpg

### CODE REVIEW

**Comment by olafurpg** [comment]
I think it's an acceptable compromise to disable these traces when a system property is enabled. We can enable this system property when users export Pants builds.

**Comment by jvican** [comment]
Note that I'm proposing to make the system property disable these traces by default. So it's not up to Pants to enable this system property.
  
I just want an escape hatch for me and people doing performance work following our Performance Guide to get this fine-grained data if they want to. It's been proven useful in the past and I'm sure it'll prove useful in the future.

**Comment by wiwa** [comment]
Thanks for feedback, updated with proposed changes.

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala [hunk 1]
```diff
@@ -26,34 +26,46 @@ final class BraveTracer private (
...
+        catch {
+          case NonFatal(t) => ()
+        }
+      }
+    } else {
+      thunk(this)
     }
   }
 
   def traceTask[T](name: String, tags: (String, String)*)(
```
Comment (olafurpg) [inline backend/src/main/scala/bloop/tracing/BraveTracer.scala]: nit: should we add a `def traceVerboseTask`? It doesn't read nicely with the `thunk` lambda on the same parameter list as the `verbose` flag. We could have a `private def traceTaskInternal` that takes all the parameters, which the public `traceTask` and `traceVerboseTask` forward to

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala [hunk 2]
```diff
@@ -78,6 +90,7 @@ object BraveTracer {
     "http://127.0.0.1:9411/api/v2/spans"
   )
   val debugTrace = Properties.propOrFalse("zipkin.trace.debug")
+  val verboseTrace = Properties.propOrFalse("zipkin.trace.verbose")
```
Comment (olafurpg) [inline backend/src/main/scala/bloop/tracing/BraveTracer.scala]: ?
```suggestion
  val isDebugEnabled = Properties.propOrFalse("zipkin.trace.debug")
  val isVerboseEnabled = Properties.propOrFalse("zipkin.trace.verbose")
```

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala (hunk 1 above)
Comment (jvican) [reply backend/src/main/scala/bloop/tracing/BraveTracer.scala]: :+1: Could be something like `private def traceTaskInternal(name: String, verbose: Boolean, tags: (String, String*))(thunk: BraveTracer => Task[T]): Task[T]`

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala (hunk 2 above)
Comment (jvican) [reply backend/src/main/scala/bloop/tracing/BraveTracer.scala]: Sounds good to me. Should we use the `bloop` prefix instead of `zipkin`?

Comment (olafurpg) [reply backend/src/main/scala/bloop/tracing/BraveTracer.scala]: Using a bloop prefix would make sense.

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala [hunk 3]
```diff
@@ -77,7 +111,8 @@ object BraveTracer {
   val zipkinServerUrl = Option(System.getProperty("zipkin.server.url")).getOrElse(
     "http://127.0.0.1:9411/api/v2/spans"
   )
-  val debugTrace = Properties.propOrFalse("zipkin.trace.debug")
+  val isDebugTrace = Properties.propOrFalse("zipkin.trace.debug")
```
Comment (olafurpg) [inline backend/src/main/scala/bloop/tracing/BraveTracer.scala]: Although this line is unrelated to the PR I think it's good to also use a `bloop.` prefix for consistency with `bloop.trace.verbose`
```suggestion
  val isDebugTrace = Properties.propOrFalse("bloop.trace.debug")
```

**Comment by jvican** [comment]
@wiwa Any chance you can have a look at the latest feedback? Should be easy to address, we can merge right after it's ready!

File: backend/src/main/scala/bloop/tracing/BraveTracer.scala (hunk 3 above)
Comment (wiwa) [reply backend/src/main/scala/bloop/tracing/BraveTracer.scala]: done!
